### PR TITLE
Support proper utf8 strings in window titles

### PR DESCRIPTION
**Pull request by t-wissmann**
This uses the XFontSet structure and XmbDrawString routines that are
capable of utf8 encoded strings. The setlocale() in the main is of great
importance, because otherwise, text drawing stops on the first non-ascii
character.

Unfortunately, XCreateFontSet() or XmbDrawString() seem to be
incompatible with the terminus font on my installation.

If the XCreateFontSet fails, then we fall back to plain XFontStruct and
XDrawString.

### CODE REVIEW

**Comment by t-wissmann** [comment]
windowtitle-x11-font-unicode

File: src/main.cpp [hunk 1]
```diff
@@ -516,6 +517,10 @@ static void sigaction_signal(int signum, void (*handler)(int)) {
 int main(int argc, char* argv[]) {
     Globals g;
     parse_arguments(argc, argv, g);
+
+    if(!setlocale(LC_CTYPE, "") || !XSupportsLocale()) {
+        fputs("warning: no locale support\n", stderr);
```
Comment (blarz) [inline src/main.cpp:522]: Why fputs when in the the rest of the main we use std::cerr?

File: src/main.cpp [hunk 2]
```diff
@@ -516,6 +517,10 @@ static void sigaction_signal(int signum, void (*handler)(int)) {
 int main(int argc, char* argv[]) {
     Globals g;
     parse_arguments(argc, argv, g);
+
+    if(!setlocale(LC_CTYPE, "") || !XSupportsLocale()) {
```
Comment (blarz) [inline src/main.cpp:521]: Nit pick: Can you add a space between if and the opening parentheses?

File: src/decoration.cpp [hunk 3]
```diff
@@ -380,13 +380,24 @@ void Decoration::redrawPixmap() {
...
-                    client_->title_().c_str(), client_->title_().size());
+    if (s.title_height() > 0) {
+        FontData& fontData = s.title_font->data();
+        string title = client_->title_();
+        Point2D titlepos = {
+            static_cast<int>(s.padding_left() + s.border_width()),
+            static_cast<int>(s.title_height())
+        };
+        if (fontData.xFontSet_) {
+            XSetForeground(g_display, gc, get_client_color(s.title_color));
```
Comment (blarz) [inline src/decoration.cpp:391]: Could be moved before the if because it is used the same way in both branches?

**Review by blarz — CHANGES_REQUESTED** [review]
Looks good, and the Firefox tabs don't have messed up characters anymore, yay ;)

Left some tiny comments

File: src/main.cpp (hunk 1 above)
Comment (t-wissmann) [reply src/main.cpp]: Because I've stole the line from dwm :'-)... It's fixed now.

File: src/main.cpp (hunk 2 above)
Comment (t-wissmann) [reply src/main.cpp]: Sure (same issue as for the `fputs`...)

File: src/decoration.cpp (hunk 3 above)
Comment (t-wissmann) [reply src/decoration.cpp:391]: It should stay there, because for xft (PR #1091) it is different.

File: src/main.cpp (hunk 1 above)
Comment (blarz) [reply src/main.cpp]: I see :D Thanks :)

File: src/main.cpp (hunk 2 above)
Comment (blarz) [reply src/main.cpp]: Thanks!

File: src/decoration.cpp (hunk 3 above)
Comment (blarz) [reply src/decoration.cpp:391]: Alright! Thanks for clearing that up

**Review by blarz — APPROVED** [review]
Looks good, thanks!

**Pull request merged**